daemon/client/conn/conn.go

```diff
@@ -101,25 +102,37 @@ func (c *conn) Echo(arg string) (string, error) {
 	return r.Pong, nil
 }
 
-func (c *conn) Run(cmd *runner.Command) (runner.ProcessStatus, error) {
+func (c *conn) Run(cmd *runner.Command) runner.ProcessStatus {
 	req := &protocol.Command{}
 	req.Argv = cmd.Argv
 	req.Env = cmd.EnvVars
 	req.Timeout = int64(cmd.Timeout)
 
 	r, err := c.client.Run(context.Background(), req)
 	if err != nil {
-		return runner.ProcessStatus{}, err
+		return runner.ProcessStatus{State: runner.FAILED, Error: err.Error()}
 	}
-	return protocol.ToRunnerStatus(r), nil
+	return protocol.ToRunnerStatus(r)
 }
 
-func (c *conn) Status(run runner.RunId) (runner.ProcessStatus, error) {
+func (c *conn) Status(run runner.RunId) runner.ProcessStatus {
 	r, err := c.client.Status(context.Background(), &protocol.StatusQuery{RunId: string(run)})
 	if err != nil {
-		return runner.ProcessStatus{}, err
+		return runner.ProcessStatus{State: runner.FAILED, Error: err.Error()}
 	}
-	return protocol.ToRunnerStatus(r), nil
+	return protocol.ToRunnerStatus(r)
+}
+
+func (c *conn) StatusAll() []runner.ProcessStatus {
```

Comment:
BTW, let me just highlight this as a place where I have no idea what the right thing to do is.

Conn wants to offer "hey, you can run this". Options for how to do that:
a) just offer grpc interface
b) create a new domain model
c) be a runner.Runner

All of these seem bad, but for different reasons. So, yeah, clumsy, and I'm sad. What do you think?

(Not a comment on this review, really, but just a conversation to keep in the back of our minds)